Tools/wasm/emscripten/__main__.py

```diff
@@ -227,8 +226,13 @@ def configure_emscripten_python(context, working_dir):
         "--enable-wasm-dynamic-linking",
         f"--prefix={PREFIX_DIR}",
     ]
-    if pydebug:
-        configure.append("--with-pydebug")
+    for flag in abiflags:
+        if flag == 'd':
+            configure.append("--with-pydebug")
+        elif flag == 't':
+            configure.append("--disable-gil")
```

Comment:
I'm not sure what the state of Emscripten on free threading will be... I'm torn on whether it's better to leave the option enabled, and just be aware that it will quite possible break, or to error out if a free threaded build is used. Maybe the former with a warning message?

Either way, we should probably handle it similar to `pydebug` - set a flag (`freethreaded`?) earlier in the process and then use that flag.

Reply:
Considering the build-python installation built by this script is specifically meant to be used in the cross-compilation process for the Emscripten build, I don't think we need remove it. Having it here means the user specifically asked for it.

Regarding the warning, I think it makes sense, but I think the build system is probably a more suitable place for it. What do you think?

Reply:
Yes - I think the build system is a better place for that warning - assuming it's even needed. Its *possible* free threaded builds might work... especially given that there's no thread support in Emscripten.